packages/experience-builder-sdk/src/blocks/editor/VisualEditorContext.tsx

```diff
@@ -115,12 +120,13 @@ export function VisualEditorContextProvider({
   // afterward, this effect fetches the respective entities.
   useEffect(() => {
     const resolveEntities = async () => {
-      setEntitiesFetched(false);
       const dataSourceEntityLinks = Object.values(dataSource || {});
-      await entityStore.fetchEntities([
-        ...dataSourceEntityLinks,
-        ...(designComponentsRegistry.values() || []),
-      ]);
+      const entityLinks = [...dataSourceEntityLinks, ...(designComponentsRegistry.values() || [])];
+      const deferredEntities = entityStore.fetchEntities(entityLinks);
+      // Only update the state and rerender when we're actually fetching something
+      if (deferredEntities === false) return;
+      setEntitiesFetched(false);
+      await deferredEntities;
```

Comment:
Whilst this gymnastics with moving the  `setEntitiesFetched()` looks cool, I couldn't see through to why this would make any difference. 

The `resolveEntities()` runs top to bottom (aka "runs to completion JS mechanism") hence whether it's on top or bottom it will be executed in the same frame of the event loop.

Reply:
Calling the function, triggers eventually async behavior which can be joined later via `await`. We make use of this trick, to first check whether the function is actually doing something async. If not, we know that it didn't fetch anything and thus we should not rerender. I added more comments to make this visible.